### PR TITLE
Better money and some styling fixes

### PR DESCRIPTION
**Pull request by TheOnly8Z**
Money entity is easier to pickup (UseTrigger and TriggerBounds), and is thrown a short distance just like KF2. If users have CSS mounted, it will show up as the CSS money prop (otherwise uses default model).

~~Classes can have initial weapons defined. Right now this defaults to the crowbar and HL2 pistol for all classes. Since classes are being developed on I decided to hold back on implementing the editor side of things (incase things will change).~~

Some of the indentation and whitespacing were fixed with glualint. If you prefer tabs over spaces, let me know and I'll adjust.

Also made all single quotes double quotes for consistency.

### CODE REVIEW

**Comment by tpan496** [comment]
I use Spaces as well so this perfectly fine.

File: gamemode/cl_init.lua [hunk 1]
```diff
@@ -247,8 +247,15 @@ net.Receive("Horde_SyncClasses", function ()
...
 end)
 
-hook.Add("HUDShouldDraw", "RemoveRetardRedScreen", function(name) 
+hook.Add("HUDShouldDraw", "RemoveRetardRedScreen", function(name)
     if (name == "CHudDamageIndicator") then
        return false
     end
+end)
+
+hook.Add("PlayerBindPress", "HelpfulBinds", function(ply, bind, pressed)
```
Comment (tpan496) [inline gamemode/cl_init.lua:256]: Would you be able to add a `Horde_` prefix in front of the hook? Trying to make things consistent. I noticed that I am missing that for my own function as well. You can change that if you want, or I will fix it myself.

File: gamemode/sh_class.lua [hunk 2]
```diff
@@ -21,65 +22,65 @@ HORDE.CreateClasses = function()
         "No bonus.\n\n",
         "Have access to all weapons except for special weapons.",
         100,
-        GetConVarNumber("horde_base_walkspeed"),
-        GetConVarNumber("horde_base_runspeed"),
-        {L="25% less damage taken.",
-        R="25% increased damage."}
+        GetConVar("horde_base_walkspeed"):GetInt(),
```
Comment (tpan496) [inline gamemode/sh_class.lua:25]: Ah, so this is the proper way to do it

File: gamemode/sh_item.lua [hunk 3]
```diff
@@ -73,12 +73,12 @@ end
...
     HORDE.CreateItem("Melee",      "Stunstick",      "weapon_stunstick",     100,  1, "Electric baton.", 
     {Medic=true, Assault=true, Heavy=true, Demolition=true, Survivor=true, Ghost=true, Engineer=true}, 10, -1)
-    HORDE.CreateItem("Melee",      "Crowbar",        "weapon_crowbar",       100,  1, "A rusty crowbar.",
+    HORDE.CreateItem("Melee",      "Crowbar",        "weapon_crowbar",       100,  0, "A rusty crowbar.",
     {Medic=true, Assault=true, Heavy=true, Demolition=true, Survivor=true, Ghost=true, Engineer=true}, 10, -1)
     HORDE.CreateItem("Melee",      "Combat Knife",   "arccw_go_melee_knife", 500,  2, "A reliable bayonet.\nRMB to deal a heavy slash.",
     {Medic=true, Assault=true, Heavy=true, Demolition=true, Survivor=true, Ghost=true, Engineer=true}, 10, -1)
 
-    HORDE.CreateItem("Pistol",     "9mm",            "weapon_pistol",     400,  2, "Combine standard sidearm.",
+    HORDE.CreateItem("Pistol",     "9mm",            "weapon_pistol",     200,  0, "Combine standard sidearm.",
```
Comment (tpan496) [inline gamemode/sh_item.lua:81]: I might change the weights on the weapons, but you can leave them like these for now.

File: gamemode/sv_economy.lua [hunk 4]
```diff
@@ -125,19 +126,18 @@ hook.Add("PlayerCanPickupWeapon", "Horde_Economy_Pickup", function (ply, wpn)
     if ply:IsNPC() then return true end
     if HORDE.items[wpn:GetClass()] then
         local item = HORDE.items[wpn:GetClass()]
-        if (ply:GetWeight() - item.weight < 0) or (not item.whitelist[ply:GetClass().name]) then
+        if (ply:GetWeight() - item.weight < 0) or (not item.whitelist[ply:GetClass().name] and not table.HasValue(HORDE.classes[ply:GetClass().name].weapons, wpn:GetClass())) then
```
Comment (tpan496) [inline gamemode/sv_economy.lua:129]: I would leave out this check here. Starter weapons should be configured in the item config to be usable for the class.

**Review by tpan496 — CHANGES_REQUESTED** [review]
Overall LGTM! A generic note: I would appreciate it if you can separate the PRs next time if they are doing different things. e.g. You can have one PR for money, and one for startup weapons.

For the money part, everything looks great. Just some small changes.

For the starter weapons part, I would prefer to merge these changes later. My idea is to have them configurable in the class config menu; it would be more than just giving players certain weapons on spawn. I will let you know when I get to the starter weapon stuff; would you be able to move those changes to some separate branch for now?

File: gamemode/cl_init.lua (hunk 1 above)
Comment (TheOnly8Z) [reply gamemode/cl_init.lua]: Done

File: entities/entities/horde_money/init.lua [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    self.Removing = false
+    self:PhysWake()
+end
+
+function ENT:StartTouch(entity)
+    if not self.Removing and entity:IsPlayer() and entity:Alive() and entity ~= self:GetOwner() then
+        self.Removing = true
+        entity:AddMoney(50)
+        entity:SyncEconomy()
+        entity:EmitSound("horde/dosh_" .. math.random(1, 4) .. ".ogg", 70, 100, 0.25)
```
[the remark this answers is not in the document]
Comment (TheOnly8Z) [reply entities/entities/horde_money/init.lua]: Done+

**Pull request merged**